crates/sim2h/src/lib.rs

```diff
@@ -179,17 +193,29 @@ impl Sim2h {
 
     /// if our listening socket has accepted any new connections, set them up
     fn priv_check_incoming_connections(&mut self) {
-        if let Ok(wss) = self.wss_recv.try_recv() {
-            let url: Lib3hUri = url::Url::from(wss.remote_url()).into();
-            let (job, outgoing_send) = ConnectionJob::new(wss, self.msg_send.clone());
-            let job = Arc::new(Mutex::new(job));
-            if let Err(error) = self.handle_incoming_connect(url.clone()) {
-                error!("Error handling incoming connection: {:?}", error);
-                return;
-            }
-            self.open_connections
-                .insert(url, (job.clone(), outgoing_send));
-            self.pool.push_job(Box::new(job));
+        for _i in 0..NUM_CONNECTION_THREADS {
+            let connection_states = self.connection_states.clone();
+            let open_connections = self.open_connections.clone();
+            let msg_send = self.msg_send.clone();
+            let pool = self.pool.clone();
+            let wss_recv = self.wss_recv.clone();
+
+            std::thread::spawn(move || loop {
```

Comment:
I guess sim2h is still using that code that panics if a thread panics... but we still might want to capture the join handle off this spawn / clean it up if / when the sim2h instance gets dropped?

Reply:
Yes, definitely. Will do in a follow up commit if this code ends up useful in the first place.